Sentry reporting + cleanup

ThaTiemsz opened the pull request:
Sentry reporting :tada: plus some cleaning up.
I'd recommend testing it first, of course :roll_eyes:

ThaTiemsz committed:
Sentry reporting + cleanup

ThaTiemsz committed:
One stupid tab, you got me there Travis

devsnek commented:
as long as you're doing this can you chain all those promises instead of nesting them

ThaTiemsz commented:
@devsnek You have write access, do it yourself 😄

devsnek commented on Utils/error_loggers.js:
```diff
@@ -1,9 +1,14 @@
+    if (fullErr !== undefined) Raven.captureException(fullErr)
+    bot.Channels.find((c) => c.name === 'bot-error').sendMessage(`Encountered an error while trying to run ${cObj.cause}.\nReturned error: \`\`\`${cObj.message}\`\`\``)
+  },
+  raven: (e) => {
```
you could add an "extra" param, which will get pushed to sentry for additional info
```js
raven: (error, extra) => Raven.captureException(error, { extra });
```

LW001 commented:
I did some testing, looks good from what I can tell.

Arzte commented on Commands/Commands/uservoice.js:
```diff
@@ -1,10 +1,10 @@
-var config = require('../../config.js')
-var Entities = require('html-entities').AllHtmlEntities
-var UVRegex = /https?:\/\/[\w.]+\/forums\/(\d{6,})-[\w-]+\/suggestions\/(\d{7,})(?:-[\w-]*)?/
+// const checker = require('../../Utils/access_checker')
```
If your moving variables like these over to ``const``'s, doesn't that make those variable program wide? So why repeat them, and as ``const`` after moving them to that?

ThaTiemsz replied:
Constants aren't program wide, they're basically variables, except you can't reassign or redeclare them.

Arzte replied:
alright, thanks for adding clarification on that.

Arzte commented on Commands/Commands/uservoice.js:
```diff
@@ -308,7 +308,7 @@ commands.info = {
               },
               timestamp: new Date(data.user.created_at)
             })
-          }).catch(() => {}) // Make bugsnag ignore it, error handling below
```
Comment is there on purpose, feel free to change it to sentry, however.

Dougley replied:
Actually no, leave this out.

Arzte commented on Commands/Commands/utils.js:
```diff
@@ -102,11 +99,11 @@ commands.stats = {
     }).catch(e => {
       msg.reply('an unexpected error occured while getting your stats, try again later.')
-      console.error(e)
+      logger.raven(e)
```
👍 thanks for cleaning this up

SirDoctors commented on Commands/Commands/queue.js:
```diff
@@ -1,17 +1,14 @@
+const checker = require('../../Utils/access_checker')
+const logger = require('../../Utils/error_loggers')
+const genlog = require('../../Utils/generic_logger')
+const config = require('../../config.js')
```
see my other comment on this.

Arzte commented on Commands/Commands/top10.js:
```diff
@@ -1,10 +1,10 @@
-var Entities = require('html-entities').AllHtmlEntities
-var config = require('../../config.js')
+const Entities = require('html-entities').AllHtmlEntities
+const config = require('../../config.js')
```
this too

SirDoctors commented on Commands/Commands/utils.js:
```diff
@@ -1,16 +1,13 @@
+// const checker = require('../../Utils/access_checker')
+const logger = require('../../Utils/error_loggers')
+const config = require('../../config.js')
+const analytics = require('../../Utils/orwell.js')
```
and these

Arzte commented on Utils/dbinit.js:
```diff
@@ -1,5 +1,5 @@
-var Dash = require('rethinkdbdash')
-var r = new Dash()
+const Dash = require('rethinkdbdash')
+const r = new Dash()
```
and this

Arzte commented on Utils/error_loggers.js:
```diff
@@ -1,9 +1,14 @@
+const Raven = require('raven')
 
-bugsnag.register(Config.discord.bugsnag)
+Raven.config(Config.discord.sentry).install()
```
reeee more config changes, reeee. 👍

Dougley replied:
Actually, change that to `Config.services.sentry`

ThaTiemsz committed:
Updates

Arzte committed:
Remove unneeded variables

These were lying in a if loop and are only used once, therefore its pretty useless to declare them as we never need them again, nor is it changed from anywhere.

Arzte committed:
change back config variables

I failed on the auto merge when I pulled tiemens' commit onto my local copy to unrevert part of his changes, this fixes that.

Arzte commented on Commands/Commands/uservoice.js:
```diff
@@ -308,7 +308,7 @@ commands.info = {
               timestamp: new Date(data.user.created_at)
             })
-          }).catch(() => {}) // Make bugsnag ignore it, error handling below
+          }).catch(logger.raven)
```
I was talking about the changing the wording in the previously there comment to reflect that it was ignored on purpose in that catch. Sorry if I was a bit misleading, however this should probably be reverted to ``.catch(() => {}) // Make sentry ignore it, error handling below``

ThaTiemsz committed:
Thing

Arzte commented on .gitignore:
```diff
@@ -54,4 +54,5 @@ channels.js
 # Roles config
 roles.js
 
+package-lock.json
```
If package-lock gen is turned off, this is probably redundent

ThaTiemsz committed:
Remove package-lock from gitignore

SirDoctors commented on Utils/error_loggers.js:
```diff
@@ -1,9 +1,12 @@
-  if (fullErr !== undefined) bugsnag.notify(fullErr)
-  bot.Channels.find((c) => c.name === 'bot-error').sendMessage(`Encountered an error while trying to run ${cObj.cause}.\nReturned error: \`\`\`${cObj.message}\`\`\``)
-}
+module.exports = {
```
Is this supposed to be changed from ``exports.log`` to module.exports?

ThaTiemsz committed:
Add suggestion url in !dupe embed

ThaTiemsz committed:
Only configure Raven in error_loggers

Arzte committed:
QOL for sentry, so errors have release and commit attatched

also enabled autobreadcrumbs, we'll see how that works out.

Dougley commented on Commands/Commands/uservoice.js:
```diff
@@ -308,7 +308,7 @@ commands.info = {
               timestamp: new Date(data.user.created_at)
             })
-          }).catch(() => {}) // Make bugsnag ignore it, error handling below
+          }).catch(() => {}) // Make sentry ignore it, error handling below
```
~~Why did you listen to doctors~~

`.catch(logger.raven)`

Dougley commented on index.js:
```diff
@@ -131,9 +110,13 @@ bot.Dispatcher.on(Events.GATEWAY_READY, () => {
 process.on('unhandledRejection', (reason, p) => {
   if (p !== null && reason !== null) {
-    bugsnag.notify(new Error(`Unhandled promise: ${require('util').inspect(p, {depth: 3})}: ${reason}`))
+    logger.raven(new Error(`Unhandled promise: ${require('util').inspect(p, {depth: 3})}: ${reason}`))
```
Do a `typeof` check to make sure `p` is not an error already, if it is, send it straight to raven.

devsnek commented:
please change the stuff so that the promises are chained

ThaTiemsz committed:
Stuff

nullpixel reviewed (commented):
[REMOVED]

SirDoctors commented:
@devsnek I'll be working on promises in another pr for the entire project, if that's all then I'll goahead and merge this.
